esp/services/ws_smc/ws_smcService.cpp

```diff
@@ -2621,3 +2605,50 @@ bool CWsSMCEx::onLockQuery(IEspContext &context, IEspLockQueryRequest &req, IEsp
     return true;
 }
 
+int CActivityInfoReader::run()
+{
+    PROGLOG("WsSMC CActivityInfoReader Thread started.");
+    unsigned int waitTimeMillies = autoRebuildMillSeconds;
+    while (!stopping)
+    {
+        if (!detached)
+        {
+            try
+            {
+                if (waitTimeMillies == (unsigned)-1)
+                {
+                    PROGLOG("WsSMC CActivityInfoReader Thread Re-started.");
+                    waitTimeMillies = autoRebuildMillSeconds;
```

Comment:
There looks like there might be are potential race conditions between something setting detached and the timer being set.
if gets !detached just before here, then it could wait indefinitely.

You don't really to change the timer, cycling around ever couple minutes (due to default auto rebuild timeout) to recheck detached would be fine, and code would be simpler.

Reply:
Will not change the timer.